### PR TITLE
MM-13846 Alert boxes to follow UX Guidelines

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Based on [Android Material Design](https://material.io/design/components/dialogs.html#anatomy) and [Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/views/alerts/)

Alert boxes should have the most common action to take by the user to the right and if a cancel option is available it should always be at the left

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13846

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Has UI changes

### CODE REVIEW

File: app/components/attachment_button.js [hunk 1]
```diff
@@ -206,11 +206,11 @@ export default class AttachmentButton extends PureComponent {
         }
     };
 
-    hasPhotoPermission = async () => {
+    hasPhotoPermission = async (source) => {
         if (Platform.OS === 'ios') {
             const {formatMessage} = this.context.intl;
             let permissionRequest;
-            const hasPermissionToStorage = await Permissions.check('photo');
+            const hasPermissionToStorage = await Permissions.check(source || 'photo');
```
Comment (hmhealey) [inline app/components/attachment_button.js]: Are these changes related to switching the order of the buttons? They seem to not be related

Comment (enahum) [reply app/components/attachment_button.js]: @hmhealey They are somewhat related, cause without this change the Alert was not even showing
